search/includes/classes/class-search.php

```diff
@@ -1309,6 +1312,71 @@ public function limit_field_limit( $field_limit ) {
 		return $field_limit;
 	}
 
+	/**
+	 * Check if meta is on allow list. If it isn't, set ep_skip_post_meta_sync to false
+	 *
+	 * @param {bool} $skip_sync The current value of whether the sync should be skipped or not
+	 * @param {WP_Post} $post The post that's attempting to be reindexed
+	 * @param {int|array} $meta_id Meta id.
+	 * @param {string} $meta_key Meta key.
+	 * @param {string} $meta_value Meta value.
+	 * @return {bool} The new value of whether the sync should be skipped or not
+	 */
+	public function filter__ep_skip_post_meta_sync( $skip_sync, $post, $meta_id, $meta_key, $meta_value ) {
+		// Respect previous skip values
+		if ( true === $skip_sync ) {
+			return true;
+		}
+
+		// If post meta allow list is disabled for this site, skip the allow list check
+		if ( defined( 'FILES_CLIENT_SITE_ID' ) ) {
+			if ( in_array( FILES_CLIENT_SITE_ID, self::DISABLE_POST_META_ALLOW_LIST, true ) ) {
+				return $skip_sync;
+			}
+		}
+
+		// If post is invalid, respect current sync value
+		if ( is_null( $post ) ) {
+			return $skip_sync;
+		}
+
+		$post_meta_allow_list = \apply_filters( 'vip_search_post_meta_allow_list', self::POST_META_DEFAULT_ALLOW_LIST, $post );
```

Comment:
Would it make sense to wrap this `apply_filters()` call into a reusable function in the Search class, to encapsulate (and de-duplicate) the logic?

Reply:
Better yet, all the code below should be de-duplicated too. If we can extract this out into something that both this function and `filter__ep_prepare_meta_data()` can use, it'll keep the core meta allow list logic DRY.